velox/exec/fuzzer/WriterFuzzer.cpp

```diff
@@ -0,0 +1,427 @@
+      TIMESTAMP(),
+  };
+
+  auto numKeys = boost::random::uniform_int_distribution<uint32_t>(1, 5)(rng_);
+  std::vector<std::string> keys;
+  for (auto i = 0; i < numKeys; ++i) {
+    keys.push_back(fmt::format("{}{}", prefix, i));
+
+    // Pick random, possibly complex, type.
+    types.push_back(vectorFuzzer_.randType(kNonFloatingPointTypes, 2));
+    names.push_back(keys.back());
+  }
+  return keys;
+}
+
+std::vector<std::string> WriterFuzzer::generatePartitionKeys(
+    const std::string& prefix,
+    std::vector<std::string>& names,
+    std::vector<TypePtr>& types) {
+  // Supported partition key  columns types
+  // According to VectorHasher::typeKindSupportsValueIds and
+  // https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java#L575
+  static const std::vector<TypePtr> kNonFloatingPointTypes{
+      BOOLEAN(), TINYINT(), SMALLINT(), INTEGER(), BIGINT(), VARCHAR()};
+
+  auto numKeys = boost::random::uniform_int_distribution<uint32_t>(1, 3)(rng_);
+  std::vector<std::string> keys;
+  for (auto i = 0; i < numKeys; ++i) {
+    keys.push_back(fmt::format("{}{}", prefix, i));
+
+    // Pick random scalar type.
+    types.push_back(vectorFuzzer_.randType(kNonFloatingPointTypes, 1));
+    names.push_back(keys.back());
+  }
+  return keys;
+}
+
+std::vector<RowVectorPtr> WriterFuzzer::generateInputData(
+    std::vector<std::string> names,
+    std::vector<TypePtr> types,
+    size_t partitionOffset) {
+  const auto size = vectorFuzzer_.getOptions().vectorSize;
+  auto inputType = ROW(std::move(names), std::move(types));
+  std::vector<RowVectorPtr> input;
+
+  // For partition keys, limit the distinct value to 4 to avoid exceeding
+  // partition number limit of 100. Since we could have up to 3 partition
+  // keys, it would generate up to 64 partitions.
+  std::vector<VectorPtr> partitionValues;
+  for (auto i = partitionOffset; i < inputType->size(); ++i) {
+    partitionValues.push_back(vectorFuzzer_.fuzz(inputType->childAt(i), 4));
+  }
+
+  auto partitionCount = inputType->size() - partitionOffset;
+  for (auto i = 0; i < FLAGS_num_batches; ++i) {
+    std::vector<VectorPtr> children;
+    for (auto j = children.size(); j < inputType->size(); ++j) {
+      if (j < partitionOffset) {
+        // For regular columns
+        children.push_back(vectorFuzzer_.fuzz(inputType->childAt(j), size));
+      } else {
+        // TODO Add other encoding support here besides DictionaryVector.
+        children.push_back(vectorFuzzer_.fuzzDictionary(
+            partitionValues.at(j - partitionOffset), size));
+      }
+    }
+    input.push_back(std::make_shared<RowVector>(
+        pool_.get(), inputType, nullptr, size, std::move(children)));
+  }
+
+  return input;
+}
+
+void WriterFuzzer::verifyWriter(
+    const std::vector<RowVectorPtr>& input,
+    const std::vector<std::string>& partitionKeys,
+    const std::string& outputDirectoryPath) {
+  auto plan = PlanBuilder()
```

Comment:
This is the default plan with values as input. It is better to also add table scan as input for testing? Thanks!